gw_eccentricity/eccDefinition.py

```diff
@@ -1285,33 +1442,217 @@ def get_good_extrema(self, pericenters, apocenters,
+        elif self.special_interp_kwargs_for_extrema["num_degree"] is None:
+            self.special_interp_kwargs_for_extrema["num_degree"] \
+                = self.special_interp_kwargs_for_extrema["denom_degree"]
+        elif self.special_interp_kwargs_for_extrema["denom_degree"] is None:
+            self.special_interp_kwargs_for_extrema["denom_degree"] \
+                = self.special_interp_kwargs_for_extrema["num_degree"]
+
+        rat_fit = get_rational_fit(
+            x, y, rational_fit_kwargs=self.special_interp_kwargs_for_extrema,
+            check_kwargs=False)
+        x_test = np.arange(x[0], x[-1], self.t[1] - self.t[0])
+        # save the degrees for checks at each step of iterations for finding
+        # the optimal degrees
+        old_num_degree = self.special_interp_kwargs_for_extrema["num_degree"]
+        old_denom_degree = self.special_interp_kwargs_for_extrema["denom_degree"]
+
+        # Check for nonmonotonicity and lower degrees if needed
+        fit_is_nonmonotonic = self.check_if_first_derivative_is_not_strictly_monotonic(
+            x_test, rat_fit(x_test), 1.0, data_name)
+        while fit_is_nonmonotonic:
+            if self.special_interp_kwargs_for_extrema["num_degree"] > 1:
+                self.special_interp_kwargs_for_extrema["num_degree"] -= 1
+            if self.special_interp_kwargs_for_extrema["denom_degree"] > 1:
+                self.special_interp_kwargs_for_extrema["denom_degree"] -= 1
+            if self.special_interp_kwargs_for_extrema["num_degree"] == 1 \
+                and self.special_interp_kwargs_for_extrema["denom_degree"] == 1:
+                debug_message(
+                    "Both numerator and denominator degrees are equal to 1 "
+                    "and cannot be lowered further.",
+                    debug_level=self.debug_level, important=False)
+                break
 
-    def interp(self, newX, oldX, oldY, allowExtrapolation=False,
-               interpolator="spline"):
-        """Get interpolated values.
+            debug_message(
+                "Lowering degrees to "
+                "num_degree="
+                f"{self.special_interp_kwargs_for_extrema['num_degree']}, "
+                "denom_degree="
+                f"{self.special_interp_kwargs_for_extrema['denom_degree']} "
+                "and retrying.", debug_level=self.debug_level, important=False)
+
+            # build new fit and check monotonicity
+            rat_fit = get_rational_fit(
+                x, y,
+                rational_fit_kwargs=self.special_interp_kwargs_for_extrema,
+                check_kwargs=False)
+            fit_is_nonmonotonic \
+                = self.check_if_first_derivative_is_not_strictly_monotonic(
+                x_test, rat_fit(x_test), 1.0, data_name)
+        # If fit with initial degree is monotonic, try increasing the degree
+        # for a better fit
+        if not fit_is_nonmonotonic:
```

Comment:
Would the code be easier to follow if it's like:
if fit_is_nonmonotonic:
   do something
else:
   do something else